p2panda-rs/src/instance/instance.rs

```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+//! Types and methods for deriving and maintaining `Instances`.
+
+use std::{collections::HashMap, convert::TryFrom};
+
+use crate::operation::{Operation, OperationValue};
+
+use super::error::InstanceError;
+
+/// The materialised view of a reduced collection of `Operations`
+#[derive(Debug, PartialEq, Default)]
+pub struct Instance(HashMap<String, OperationValue>);
+
+impl Instance {
+    /// Instantiate a new `Instance`
+    fn new() -> Self {
+        Self(HashMap::new())
+    }
+
+    /// Update this `Instance` from an UPDATE `Operation`
+    pub fn update(&mut self, operation: Operation) -> Result<(), InstanceError> {
```

Comment:
Another good name for this method could be `process_update` or `apply_update`

Reply:
I like these names!!